allennlp/common/push_to_hf.py

```
@@ -0,0 +1,145 @@
+
+
+_ALLOWLIST_PATHS = ["vocabulary", "config.json", "weights.th", "best.th", "metrics.json", "log"]
+
+
+def _copy_allowed_file(filepath: Path, dst_directory: Path):
+    """
+    Copies files from allowlist to a directory, overriding existing
+    files or directories if any.
+    """
+    if filepath.name not in _ALLOWLIST_PATHS:
+        return
+
+    dst = dst_directory / filepath.name
+    if dst.is_dir():
+        shutil.rmtree(str(dst))
+    elif dst.is_file():
+        dst.unlink()
+    if filepath.is_dir():
+        shutil.copytree(filepath, dst)
+    elif filepath.is_file():
+        if filepath.name == "best.th":
+            dst = dst_directory / "model.th"
+        shutil.copy(str(filepath), str(dst))
+
+
+def push_to_hf(
+    archive_path: Union[str, Path],
+    repo_name: str,
+    organization: Optional[str] = None,
+    commit_message: str = "Update repository",
+    local_repo_path: Union[str, Path] = "hub",
+):
+    """Pushes model and related files to the Hugging Face Hub ([hf.co](https://hf.co/))
+
+    # Parameters
+
+    archive_path : `Union[str, Path]`
+        Full path to the zipped model (e.g. model/model.tar.gz) or to a directory with the serialized model.
+
+    repo_name: `str`
+        Name of the repository in the Hugging Face Hub.
+
+    organization : `Optional[str]`, optional (default = `None`)
+        Name of organization to which the model should be uploaded.
+
+    commit_message: `str` (default=`Update repository`)
+        Commit message to use for the push.
+
+    local_repo_path : `Union[str, Path]`, optional (default=`hub`)
+        Local directory where the repository will be saved.
+
+    """
+    archive_path = Path(archive_path)
+
+    if not archive_path.exists():
+        logging.error(
+            f"Can't find archive path: {archive_path}, please"
+            "point to either a .tar.gz archive or to a directory"
+            "with the serialized model."
+        )
+        sys.exit(1)
```

This should throw an exception instead. `push_to_hf()` would be a useful thing to call from an ipython command line, a Jupyter notebook, etc., and it should not break those things when the archive doesn't exist.

---

Yes, you're right. I moved this originally from the command line script, but you're right this should be an error.